cl/corpus_importer/management/commands/harvard_merge.py

```diff
@@ -0,0 +1,895 @@
+        update_dict["case_name"] = harvard_case_name
+        # Change stored value to new
+        cluster_case_name = harvard_case_name
+    elif cluster_case_name and harvard_case_name:
+        if len(harvard_case_name) > len(cluster_case_name):
+            # Select best case name based on string length
+            update_dict["case_name"] = harvard_case_name
+            # Change stored value to new
+            cluster_case_name = harvard_case_name
+    else:
+        # We don't care if harvard data is empty or both are empty
+        pass
+
+    if cluster_case_name_full and cluster_case_name:
+        if len(cluster_case_name) > len(cluster_case_name_full):
+            # Swap field values
+            update_dict["case_name"] = cluster_case_name_full
+            update_dict["case_name_full"] = cluster_case_name
+
+    if update_dict:
+        OpinionCluster.objects.filter(id=cluster_id).update(**update_dict)
+
+
+def merge_date_filed(cluster_id: int, harvard_data: dict) -> None:
+    """Merge date filed
+
+    :param cluster_id: The cluster id of the merging item
+    :param harvard_data: json data from harvard case
+    :return: None
+    """
+    cluster = OpinionCluster.objects.get(id=cluster_id)
+    harvard_date_filed = harvard_data.get("decision_date")
+    cluster_date_filed = cluster.date_filed
+    merge_cluster_dates(
+        cluster_id, "date_filed", (harvard_date_filed, cluster_date_filed)
+    )
+
+
+def merge_overlapping_data(
+    cluster_id: int, changed_values_dictionary: dict
+) -> None:
+    """Merge overlapping data
+
+    :param cluster_id: the cluster id
+    :param changed_values_dictionary: the dictionary of data to merge
+    :return: None
+    """
+
+    if not changed_values_dictionary:
+        # Empty dictionary means that we don't have overlapping data
+        return
+
+    long_fields = [
+        "syllabus",
+        "summary",
+        "history",
+        "headnotes",
+        "correction",
+        "cross_reference",
+        "disposition",
+        "arguments",
+    ]
+
+    for field_name in changed_values_dictionary.keys():
+        if field_name in long_fields:
+            merge_long_fields(
+                cluster_id,
+                field_name,
+                changed_values_dictionary.get(field_name),
+            )
+        elif field_name in ["other_dates"]:
+            merge_cluster_dates(
+                cluster_id,
+                field_name,
+                changed_values_dictionary.get(field_name),
+            )
+        elif field_name == "judges":
+            merge_judges(
+                cluster_id,
+                changed_values_dictionary.get(field_name),
+            )
+        elif field_name == "attorneys":
+            merge_strings(
+                cluster_id,
+                field_name,
+                changed_values_dictionary.get(field_name, ""),
+            )
+        else:
+            logger.info(f"Field not considered in the process: {field_name}")
```

Comment:
Should this raise a NotImplemented exception, maybe? We're expecting to never hit this, right?

Reply:
Ideally we do not expect to reach this condition, it is just in case there is some field that we are not aware of.